internal/provider/indices_datasource.go

```diff
@@ -390,6 +498,18 @@ func (d *indicesDataSource) Read(ctx context.Context, req datasource.ReadRequest
 			FilterStringMaxLength: types.StringValue(fmt.Sprintf("%d", indexDetail.FilterStringMaxLength)),
 		}
 
+		// Handle model properties
+		if items[i].ModelProperties.Name.IsNull() &&
```

Comment:
Nit: you could make this a method on the ModelProperties struct since it logically belongs to that type:
```go
func (m ModelPropertiesModel) IsEmpty() bool {
	return m.Name.IsNull() &&
		m.Dimensions.IsNull() &&
		m.Type.IsNull() &&
		m.Tokens.IsNull() &&
		m.Url.IsNull() &&
		m.TrustRemoteCode.IsNull() &&
		m.IsMarqtunedModel.IsNull()
}
```

Reply:
Thanks, that's cleaner

Reply:
modified a bit and added